accounting/job_archive_interface.py

```diff
@@ -274,26 +274,47 @@ def calc_usage_factor(
     # that determine how far back to go in the job-archive
     after_start_time = priority_decay_half_life * 604800
 
+    # fetch timestamp of last seen job (will fetch jobs that
+    # have run after this time)
+    fetch_timestamp_query = """
+        SELECT last_job_timestamp
+        FROM job_usage_factor_table
+        WHERE username=? AND bank=?
+        """
+    dataframe = pd.read_sql_query(
+        fetch_timestamp_query, acct_conn, params=(user, bank,)
+    )
+    timestamp = dataframe.iloc[0]
+
     # get the total number of nodes and time elapsed across
-    # all of a user's jobs that completed before an end time;
-    # just store job records in a temporary .csv file that gets
-    # deleted when the calculation are done.
-    output_file = "tmp.csv"
+    # all of a user's jobs that completed since the last completed
+    # job that was retrieved
     job_totals_user = view_job_records(
-        jobs_conn,
-        output_file,
-        user=user,
-        after_start_time=(time.time() - after_start_time),
+        jobs_conn, output_file=None, user=user, after_start_time=float(timestamp),
     )
 
-    total_nodes_user = 0
-    total_t_elapsed_user = 0
+    # one per job factor = nnodes * t_elapsed
+    # usage factors = total(nnodes * t_elapsed)
+    per_job_factors = []
+    last_t_inactive = 0.0
     for job in job_totals_user:
-        total_nodes_user += job.nnodes
-        total_t_elapsed_user += job.elapsed
+        per_job_factors.append(round((job.nnodes * job.elapsed), 5))
+        last_t_inactive = job.t_inactive
 
-    # current usage factor = total num of nodes * total time elapsed of jobs
-    usage_user_current = total_nodes_user * total_t_elapsed_user
+    usage_user_current = sum(per_job_factors)
+
+    # write last t_inactive to last_job_timestamp for user
+    update_timestamp_stmt = (
+        "UPDATE job_usage_factor_table SET last_job_timestamp="
+        + str(last_t_inactive)
+        + " WHERE username='"
+        + user
+        + "' AND bank='"
+        + bank
+        + "'"
+    )
+    acct_conn.execute(update_timestamp_stmt)
```

Comment:
Rather than using string formatting or string concatenation, I'd recommend this use the parameter substitution supported by the `execute` funcion:

```python
    update_timestamp_stmt = (
        "UPDATE job_usage_factor_table SET last_job_timestamp=?" \
        " WHERE username='?' AND bank='?'"
    )
    acct_conn.execute(update_timestamp_stmt, (user, bank))
```

there is also a named variant:

```python
    update_timestamp_stmt = (
        "UPDATE job_usage_factor_table SET last_job_timestamp=?" \
        " WHERE username=':name' AND bank=':bank'"
    )
    acct_conn.execute(update_timestamp_stmt, {"user": user, "bank": bank})
```

Reply:
Same comment for the `update_stmt` below